src/client/components/loginSignUpPage/SignUpPage.jsx

```diff
@@ -1,76 +1,124 @@
 import styled from 'styled-components';
-import { OrangeButton } from '/src/client/styles/shared.tsx';
 import axios from 'axios';
+import { Button, Icon, Form } from 'react-bulma-components';
 import { useHistory } from 'react-router-dom'
 import { OrangeNavbar } from '../../styles/shared';
+import { OrangeButton } from '/src/client/styles/shared.tsx';
+import validate from './validationRules.js';
+import useForm from './formValidation.js';
 
 const StyledHeader = styled.h2`
 font-size: 20px;
 font-weight: bold;
+margin-bottom: 5px;
 `;
 
 export default function SignUpPage() {
-  const [firstName, setFirstName] = useState('');
-  const [lastName, setLastName] = useState('');
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-
-  const handleSignUp = () => {
-    axios.post('', {
-      //object to send based on database structure
-    })
-    .then(res => {
-      console.log('successfully registered user');
-      //redirect user to the login page?
-    })
-    .catch(err => {
-      console.log(err)
-    })
-  }
+  const [username, setUsername] = useState('');
 
   const history = useHistory();
   function handleHomeClick() {
     history.push("/");
   }
+
   const goBack = () => {
     history.goBack()
   }
 
+  const signUp = () => {
+    axios.post('/create', {
+      first_name: values.firstName,
+      last_name: values.lastName,
+      email: values.email,
+      username: username,
+      password: values.password,
+      guest: false
+    })
+      .then(res => {
+        console.log('successfully registered user');
+        goBack();
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  }
+
+  const { values, handleChange, errors, handleSubmit } = useForm(signUp, validate);
+
   return (
     <div className="login-signup-background">
-      < OrangeNavbar needBackArrow={true} onBackArrowClick={goBack}/>
+      < OrangeNavbar needBackArrow={true} onBackArrowClick={goBack} />
       <div className="login-signup-page-container">
         <StyledHeader>New Account</StyledHeader>
         <Form.Field className="login-form">
           <Form.Label>First Name</Form.Label>
-          <Form.Control>
-            <Form.Input placeholder="First Name" onChange={e => {setFirstName(e.target.value)}}/>
+          <Form.Control className="form-spacing-signup">
+            <Form.Input
+              name="firstName"
+              className={`input ${errors.firstName && "is-danger"}`}
+              placeholder="First Name"
+              onChange={handleChange}
+            />
+            {errors.email && (
+              <Form.Help className="help is-danger">{errors.firstName}</Form.Help>
+            )}
           </Form.Control>
           <Form.Label>Last Name</Form.Label>
-          <Form.Control>
-            <Form.Input placeholder="Last Name" onChange={e => {setLastName(e.target.value)}}/>
+          <Form.Control className="form-spacing-signup">
```

Comment:
I see lots of opportunities to create reusable Form.Control components and condensing this return down to a fraction of its size, but if you think it will take too long, just create a tech-debt ticket and we can move on

Reply:
yeah it's pretty messy with the form stuff so i'll revise once we get the other stuff down